src/lxc/criu.c

```diff
@@ -363,10 +363,11 @@ static void exec_criu(struct criu_opts *opts)
 	argv[argc] = NULL;
 
 	buf[0] = 0;
-	pos = buf;
+	pos = 0;
 	for (i = 0; argv[i]; i++) {
-		pos = strncat(buf, argv[i], buf + sizeof(buf) - pos);
-		pos = strncat(buf, " ", buf + sizeof(buf) - pos);
+		strncat(buf, argv[i], sizeof(buf) - pos - 1);
```

Comment:
Also every strncat and strlen is walking the length of that string.  It would
be nice to just do something like:

	size_t pos = 0;
	for (i = 0; i < arv[i]; i++) {
		ret = snprintf(buf + pos, sizeof(buf) - pos, "%s", argv[i]);
		if (ret < 0 || ret + pos >= sizeof(buf))
			EOVERFLOW
		pos += ret;
	}